apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveService.kt

```diff
@@ -228,53 +224,85 @@ class OppgaveService(
         }
     }
 
-    fun ferdigstillOppgaveUnderbehandlingOgLagNyMedType(
-        fattetoppgaveReferanseOgSak: SakIdOgReferanse,
-        oppgaveType: OppgaveType,
+    fun tilAttestering(
+        referanse: String,
+        type: OppgaveType,
         merknad: String?,
-        saksbehandler: BrukerTokenInfo,
+        frist: Tidspunkt? = null,
     ): OppgaveIntern {
-        val behandlingsoppgaver = oppgaveDao.hentOppgaverForReferanse(fattetoppgaveReferanseOgSak.referanse)
-        if (behandlingsoppgaver.isEmpty()) {
-            throw BadRequestException("Må ha en oppgave for å kunne lage attesteringsoppgave")
+        val oppgaver =
+            hentOppgaverForReferanse(referanse)
+                .filter { it.type == type }
+                .filter { it.status in listOf(Status.UNDER_BEHANDLING, Status.UNDERKJENT) }
+
+        if (oppgaver.isEmpty()) {
+            throw ManglerOppgaveUnderBehandling("Ingen oppgave funnet for referanse: $referanse")
+        } else if (oppgaver.size > 1) {
+            throw ForMangeOppgaverUnderBehandling("For mange oppgaver under behandling (ref: $referanse)")
         }
-        try {
-            val oppgaveUnderbehandling = behandlingsoppgaver.single { it.status == Status.UNDER_BEHANDLING }
-            ferdigstillOppgaveById(oppgaveUnderbehandling, saksbehandler)
-            return opprettNyOppgaveMedSakOgReferanse(
-                referanse = fattetoppgaveReferanseOgSak.referanse,
-                sakId = fattetoppgaveReferanseOgSak.sakId,
-                oppgaveKilde = oppgaveUnderbehandling.kilde,
-                oppgaveType = oppgaveType,
-                merknad = merknad,
-            )
-        } catch (e: NoSuchElementException) {
-            throw BadRequestException(
-                "Det må finnes en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
-        } catch (e: IllegalArgumentException) {
-            throw BadRequestException(
-                "Skal kun ha en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
+
+        val oppgave = oppgaver.single()
```

Comment:
nitpick men kan vel kalle `first` her, eller er det ekstra sikring?

Reply:
Det skal kun være én oppgave per referanse med status `under behandling` eller `underkjent` for å endre status til `attestering´. Så tenker single er mest logisk. First er litt risky siden det bare tar første og beste, som ikke nødvendigvis blir riktig.